open_tms_backend/open_tms_delete_department/index.js

```diff
@@ -0,0 +1,212 @@
+ *     parameters:
+ *       - name: id
+ *         in: query
+ *         description: Department ID to delete
+ *         required: true
+ *         type: string
+ *     responses:
+ *       200:
+ *         description: Department deleted successfully.
+ *         content:
+ *           application/json:
+ *             example:
+ *               meta:
+ *                 status: 200
+ *                 message: Department deleted successfully
+ *       400:
+ *         description: Bad Request. Invalid input or missing fields.
+ *         content:
+ *           application/json:
+ *             example:
+ *               meta:
+ *                 status: 400
+ *                 message: Bad Request
+ *               errors:
+ *                 - code: BAD_REQUEST
+ *                   message: Please provide a valid department id
+ *       404:
+ *         description: Bad Request. Record to delete does not exist.
+ *         content:
+ *           application/json:
+ *             example:
+ *               meta:
+ *                 status: 404
+ *                 message: Bad Request
+ *               errors:
+ *                 - code: RESOURCE_NOT_FOUND
+ *                   message: Record to delete does not exist.
+ *       422:
+ *         description: Department has members.
+ *         content:
+ *           application/json:
+ *             example:
+ *               meta:
+ *                 status: 422
+ *                 message:Unprocessable Entity
+ *               errors:
+ *                 - code: DEPARTMENT_HAS_MEMBERS
+ *                   message: The department cannot be deleted as it contains active members.
+ *       500:
+ *         description: Internal Server Error. Something went wrong.
+ *         content:
+ *           application/json:
+ *             example:
+ *               meta:
+ *                 status: 500
+ *                 message: Something went wrong
+ *               errors:
+ *                 - code: INTERNAL_ERROR
+ *                   message: Something went wrong
+ */
+const handler = async (event) => {
+  const { req, res } = event
+  const { sendResponse, prisma, validateRequestMethod, HTTP_STATUS_CODES } = await shared.getShared()
+
+  // health check
+  if (req.params.health === 'health') {
+    return sendResponse(res, 200, { message: 'Health check success' })
+  }
+
+  await validateRequestMethod(req, ['DELETE'])
+
+  // check if req contains id param
+
+  // if empty throw error
+  if (!req.query?.id) {
+    return sendResponse(res, 400, {
+      meta: {
+        status: 400,
+        message: HTTP_STATUS_CODES[400],
+      },
+      errors: [
+        {
+          code: 'MISSING_ID',
+          message: `department id to delete is missing in request`,
+        },
+      ],
+    })
+  }
+
+  const { id } = req.query
+
+  try {
+    // check if the department has any users associated with it
+    const departmentUsers = await prisma.department_users.findFirst({
```

Comment:
No need to check if the department has any users.We can also delete department which has existing users